### PR TITLE
BRC20 token amount holding

### PR DESCRIPTION
**Pull request by BillyWooo**
### Context

Add BRC20 token amount holding VC.
This one VC will include all related token info from list: "ordi", "sats", "rats", "MMSS", "long", "cats", "BTCs".

### Labels

Please apply following PR-related labels when appropriate:
- `C0-breaking`: if your change could break the existing client, e.g. API change, critical logic change 
- `C1-noteworthy`: if your change is non-breaking, but is still worth noticing for the client, e.g. reference code improvement

### Testing Evidences

Please attach any relevant evidences if applicable

### CODE REVIEW

File: tee-worker/litentry/core/credentials/src/brc20/amount_holder.rs [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+fn get_assertion_content(token: &BRCToken) -> &'static str {
+	match token {
+		BRCToken::ORDI => "$ordi_holding_amount",
+		BRCToken::SATS => "$sats_holding_amount",
+		BRCToken::RATS => "$rats_holding_amount",
+		BRCToken::MMSS => "$MMSS_holding_amount",
+		BRCToken::LONG => "$long_holding_amount",
+		BRCToken::CATS => "$cats_holding_amount",
+		BRCToken::BTCS => "$BTCs_holding_amount",
```
Comment (wli-pro) [inline tee-worker/litentry/core/credentials/src/brc20/amount_holder.rs]: Actually, can we not create new VC with $ordi_holding_amount but put token into seperate Assertion condition

File: primitives/core/src/assertion.rs [hunk 2]
```diff
@@ -246,6 +246,9 @@ pub enum Assertion {
 	#[codec(index = 19)]
 	WeirdoGhostGangHolder,
 
+	#[codec(index = 20)]
+	BRC20AmountHolder,
+
```
Comment (outofxxx) [inline primitives/core/src/assertion.rs]: As we discussed earlier, `LIT staking` ' index is 20, `index = 23` is available for `BRCAmountHolder`

File: tee-worker/litentry/core/assertion-build/src/brc20/amount_holder.rs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+use lc_data_providers::{
+	geniidata::GeniidataClient, DataProviderConfigReader, ReadDataProviderConfig,
+};
+
+pub fn build(req: &AssertionBuildRequest) -> Result<Credential> {
+	let identities = transpose_identity(&req.identities);
+
+	// TODO: here still missing one step: convert 'address', which is a pubkey, into real BTC/BRC20 address,
+	// which the data provider (GeniiData.com) can recognize.
+	let addresses = identities
```
Comment (outofxxx) [inline tee-worker/litentry/core/assertion-build/src/brc20/amount_holder.rs]: `GeniiData` doesn't support pubkey?
So what format of address should be converted here？

**Review by outofxxx — COMMENTED** [review]
And don't forget to update vc definition.

**Pull request merged**